server/eclipse-project/src/main/java/us/freeandfair/corla/model/CountyDashboard.java

```diff
@@ -735,82 +731,109 @@ public void setBallotsInManifest(final Integer the_ballots_in_manifest) {
    * @param the_map The map.
    * @param the_key The key.
    */
-  private void ensureCounterExists(final Map<AuditReason, Integer> the_map,
-                                   final AuditReason the_key) {
+  private void ensureCounterExists(final Map<AuditSelection, Integer> the_map,
+                                   final AuditSelection the_key) {
     if (!the_map.containsKey(the_key)) {
       the_map.put(the_key, 0);
     }
   }
   
   /**
    * @return the numbers of discrepancies found in the audit so far, 
-   * categorized by contest audit reason.
+   * categorized by contest audit selection.
    */
-  public Map<AuditReason, Integer> discrepancies() {
+  public Map<AuditSelection, Integer> discrepancies() {
     return Collections.unmodifiableMap(my_discrepancies);
   }
   
   /**
-   * Adds a discrepancy for the specified audit reason. This adds it both to the 
+   * Adds a discrepancy for the specified audit reasons. This adds it both to the 
    * total and to the current audit round, if one is ongoing.
    * 
-   * @param the_reason The reason.
+   * @param the_reasons The reasons.
    */
-  public void addDiscrepancy(final AuditReason the_reason) {
-    ensureCounterExists(my_discrepancies, the_reason);
-    my_discrepancies.put(the_reason, my_discrepancies.get(the_reason) + 1); 
+  public void addDiscrepancy(final Set<AuditReason> the_reasons) {
+    final Set<AuditSelection> selections = new HashSet<>();
+    for (final AuditReason r : the_reasons) {
+      selections.add(r.selection());
+    }
+    for (final AuditSelection s : selections) {
+      ensureCounterExists(my_discrepancies, s);
+      my_discrepancies.put(s, my_discrepancies.get(s) + 1);
```

Comment:
Could you use [`getOrDefault()`](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#getOrDefault-java.lang.Object-V-) and get rid of `ensureCounterExists()` altogether?

Reply:
Also apply this comment to the `*Disagreement()` methods.